manifests/0000_90_kube-apiserver-operator_04_servicemonitor-apiserver.yaml

```diff
@@ -135,3 +135,16 @@ spec:
     - record: cluster:apiserver_current_inflight_requests:sum:max_over_time:2m
       expr: |
         max_over_time(sum(apiserver_current_inflight_requests{apiserver=~"openshift-apiserver|kube-apiserver"}) by (apiserver,requestKind)[2m:])
+  - name: pre-release-lifecycle
+    rules:
+      - alert: DeprecatedAPIInUse
+        annotations:
+          message: >-
+            Deprecated API that will be removed in the next version is being used. Removing the workload that is using
+            the {{"{{$labels.group}}"}}.{{"{{$labels.version}}"}}/{{"{{$labels.resource}}"}} API might be necessary for
+            a successful upgrade to the next cluster version. Refer to the audit logs to identify the workload.
+        expr: |
+          group(apiserver_requested_deprecated_apis{removed_release="1.21"}) by (group,version,resource) and (sum by(group,version,resource) (rate(apiserver_request_total[10m]))) > 0
+        for: 1h
```

Comment:
if we use 4h, will that cover the entire timeframe of an e2e test?  Is this the spot or the rate the spot?  Probably here I guess.

Reply:
Followed up at https://github.com/openshift/cluster-kube-apiserver-operator/pull/1044.